File: 2015/1/1.go
package main

import (
	"fmt"
	"os"
)

func parseFile(filename string) string{
	file, err := os.ReadFile(filename)
	if err != nil {
		fmt.Printf("Failed to open file: %v", err)
	}

	text := string(file)
	return text
}

func checkFloor(text string) int {
	floor := 0
	for _, char := range text {
		if char == '(' {
			floor++
		} else {
			floor--
		}
	}
	return floor
}

func firstChance(text string) int {
	floor := 0
	for pos, char := range text {
		if char == '(' {
			floor++
		} else {
			floor--
			if floor == -1 {
				return pos + 1
			}
		}
	}
	return -1
}

func main() {
	filename := "1.txt"
	text := parseFile(filename)
	ans1 := checkFloor(text)
	ans2 := firstChance(text)
	fmt.Println("Answer for part 1: ", ans1)
	fmt.Println("Answer for part 2: ", ans2)
}	

File: 2015/1/1.py


def parse_file(filename):
    text = ""
    with open(filename, 'r') as file:
        for line in file:
            text = line.strip()
    return text

def floor_counter(text):
    n = len(text)
    floor = 0
    for i in range(n):
        if "(" == text[i]:
            floor += 1
        else:
            floor -= 1
    return floor
    
def first_chance(text):
    n = len(text)
    floor = 0
    for i in range(n):
        if "(" == text[i]:
            floor += 1
        else:
            floor -= 1
            if floor == -1:
                return i
    return -1


text = parse_file("1.txt")
ans1 = floor_counter(text)
ans2 = first_chance(text) + 1 # This is not considered as a 0 index array
print("Answer for part 1: ", ans1)
print("Answer for part 2: ", ans2)